src/PostgREST/QueryBuilder.hs

```diff
@@ -339,7 +382,7 @@ requestToQuery schema isParent (DbRead (Node (Select colSelects tbls conditions
     --the following is just to remove the warning
     --getQueryParts is not total but requestToQuery is called only after addJoinConditions which ensures the only
     --posible relations are Child Parent Many
-    getQueryParts (Node (_,(_,Nothing,_)) _) _ = undefined
+    getQueryParts _ _ = undefined --error "undefined getQueryParts"
```

Comment:
this specific case is never reached since this function does not get called when the previous code cant find a relation between to nodes, and this undefined part is here to match a node where the relation type is `Nothing`